gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/diagnosticreport/SpecimenCompoundsMapper.java

```diff
@@ -214,32 +214,27 @@ private void handleClusterCompoundStatement(RCMRMT030101UKCompoundStatement spec
     }
 
     private void handleBatteryCompoundStatement(RCMRMT030101UKCompoundStatement specimenCompoundStatement,
-        RCMRMT030101UKCompoundStatement batteryCompoundStatement,
+                                                RCMRMT030101UKCompoundStatement batteryCompoundStatement,
+                                                List<Observation> observations,
+                                                List<Observation> observationComments,
+                                                DiagnosticReport diagnosticReport) {
 
-        List<Observation> observations, List<Observation> observationComments, DiagnosticReport diagnosticReport) {
-
-        var compoundStatements = batteryCompoundStatement.getComponent().stream()
+        batteryCompoundStatement.getComponent().stream()
             .filter(RCMRMT030101UKComponent02::hasCompoundStatement)
             .map(RCMRMT030101UKComponent02::getCompoundStatement)
             .filter(compoundStatement -> CLUSTER_CLASSCODE.equals(compoundStatement.getClassCode().get(0)))
-            .toList();
-
-        for (var compoundStatement : compoundStatements) {
-            handleClusterCompoundStatement(
-                specimenCompoundStatement, compoundStatement, observations, observationComments, diagnosticReport, true
-            );
-        }
+            .forEach(compoundStatement ->
+                         handleClusterCompoundStatement(
+                             specimenCompoundStatement, compoundStatement, observations, observationComments, diagnosticReport, true
+                                                       ));
 
-        var observationStatements = batteryCompoundStatement.getComponent().stream()
+        batteryCompoundStatement.getComponent().stream()
             .filter(RCMRMT030101UKComponent02::hasObservationStatement)
             .map(RCMRMT030101UKComponent02::getObservationStatement)
-            .toList();
+            .forEach(
+                observationStatement -> getObservationById(observations, observationStatement.getId().getRoot()).ifPresent(
+                    observation -> handleObservationStatement(specimenCompoundStatement, observationStatement, observation)));
```

Comment:
Similarly, can we try formatting this a little - i.e. 

```suggestion
            .forEach(observationStatement -> getObservationById(
                observations, 
                observationStatement.getId().getRoot())
                .ifPresent(observation -> handleObservationStatement(
                    specimenCompoundStatement, 
                    observationStatement, 
                    observation)
                )
            );
```